### PR TITLE
Prevent double free for `CMS_add1_cert()` and fix doc - backport to 3.0 and 3.1

### PR DESCRIPTION
**Pull request by DDvO**
Backport the parts of #19199 containing fixes for `CMS_add1_cert()`.

### CODE REVIEW

File: doc/man3/CMS_add0_cert.pod [hunk 1]
```diff
@@ -48,8 +51,7 @@ As the I<0> implies CMS_add0_cert() adds I<cert> internally to I<cms> and it
 must not be freed up after the call as opposed to CMS_add1_cert() where I<cert>
 must be freed up.
 
-The same certificate or CRL must not be added to the same cms structure more
-than once.
+The same certificate must not be added to the same cms structure more than once.
```
Comment (slontis) [inline doc/man3/CMS_add0_cert.pod:54]: Why the change here?

Comment (DDvO) [reply doc/man3/CMS_add0_cert.pod:54]: This change fixes the documentation.
The code does not implement a uniqueness restriction on CRLs.

**Review by paulidale — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by DDvO** [comment]
Thank you for the approvals.

Interesting that this backport has been approved before the original PR: #19199.

**Comment by DDvO** [comment]
Merged to 3.0 and cherry-picked to 3.1 - thanks @paulidale and @hlandau for the approvals.
